### DISCUSSION TITLE
Playerbots module (ZhengPeiRu21)  not compiling correctly

### DISCUSSION THREAD

**Discussion by skrissz92** [azerothcore/azerothcore-wotlk#16016] [Modules]
I have been trying to get the ZhengPeiRu21 playerbot module (https://github.com/ZhengPeiRu21/mod-playerbots) to compile without success. I use the correct tree of azerothcore as described in the readme.md for this: https://github.com/ZhengPeiRu21/azerothcore-wotlk/tree/Playerbot

examples errors:

- 22>F:\azerothcore\modules\mod-playerbots\src\strategy\actions\DebugAction.cpp(824,27): error C2660: 'Unit::SendPlaySpellVisual': function does not take 2 arguments
- 22>F:\azerothcore\src\server\game\Entities\Unit\Unit.h(1757,10): message : see declaration of 'Unit::SendPlaySpellVisual' (compiling source file F:\azerothcore\modules\mod-playerbots\src\strategy\actions\DebugAction.cpp)
- 22>F:\azerothcore\modules\mod-playerbots\src\strategy\actions\DebugAction.cpp(824,27): message : while trying to match the argument list '(ObjectGuid, int)'
- 
- 22>DestroyItemAction.cpp
- 22>F:\azerothcore\src\server\game\Battlegrounds\Zones\BattlegroundAV.h(1617,19): message : see declaration of 'BattlegroundAV'
- 22>F:\azerothcore\modules\mod-playerbots\src\strategy\actions\BattleGroundTactics.cpp(2884,36): error C2039: 'GetAVNodeInfo': is not a member of 'BattlegroundAV'
- 22>F:\azerothcore\src\server\game\Battlegrounds\Zones\BattlegroundAV.h(1617,19): message : see declaration of 'BattlegroundAV'
- 22>F:\azerothcore\modules\mod-playerbots\src\strategy\actions\BattleGroundTactics.cpp(2901,66): error C2039: 'GetAVNodeInfo': is not a member of 'BattlegroundAV'
- 
- 22>F:\azerothcore\modules\mod-playerbots\src\ChatFilter.cpp(166,47): error C2039: 'GetTargetIcon': is not a member of 'Group'
- 22>F:\azerothcore\src\server\game\Entities\Player\Player.h(59,7): message : see declaration of 'Group'
- 22>F:\azerothcore\modules\mod-playerbots\src\RandomItemMgr.h(20,6): warning C5257: 'EquipmentSlots': enumeration was previously declared without a fixed underlying type (compiling source file F:\azerothcore\modules\mod-playerbots\src\GuildTaskMgr.cpp)
- 22>F:\azerothcore\modules\mod-playerbots\src\GuildTaskMgr.cpp(504,5): error C2065: 'PlayerbotsDatabasePreparedStatement': undeclared identifier
- 22>F:\azerothcore\modules\mod-playerbots\src\GuildTaskMgr.cpp(504,42): error C2065: 'stmt': undeclared identifier
- 22>F:\azerothcore\modules\mod-playerbots\src\GuildTaskMgr.cpp(504,49): error C2065: 'PlayerbotsDatabase': undeclared identifier
- 
Finally error:
- 25>LINK : fatal error LNK1181: cannot open input file '..\..\..\modules\RelWithDebInfo\modules.lib'
- 25>Done building project "worldserver.vcxproj" -- FAILED.

The Cmake process runs without issues, but when I run my compilation I a lot of errors on the playerbot module.

    Compile and run the core without modules without issues.
    Compile and run the core with compiles other modules without issues.

my steps:

    git clone https://github.com/ZhengPeiRu21/azerothcore-wotlk.git
    move to the modules folder of the downloaded code:
    git clone https://github.com/ZhengPeiRu21/mod-playerbots.git
    Run Cmake
    click configure
    set TOOLS-BUILD to all
    set MODULE-MOD-PLAYERBOTS to static
    click configure again (no errors)
    click generate
    click open project
    VS 2022 opens on the project
    set project to RelWithDebInfo and X64
    right click BUILD-ALL > clean
    right click BUILD-ALL > Build

The build works fine without the playerbot module and the world server launcher is created.
what could be the cause of the error? would I miss a step?
Any assistance would be greatly appreciated.

Maybe there is another way to install the sticks?

**Comment by Kitzunu** [comment]
Make an issue on the module's repository.

**Comment by vnasing** [comment]
使用任何模块都会出现这个错误。不单独是playerbots。能解决的大哥，帮忙解决一下
